### PR TITLE
Enhancement to add excel formula options (#1643)

### PR DESCRIPTION
**Pull request by speckyspooky**
Enhancement to add excel formula options (#1643)

### CODE REVIEW

**Comment by speckyspooky** [comment]
I researched the original changes of JT and compared it with the latest version of our master-branch.
The change is no copy or direct merge of the change. It is focused on the formula-scope.
So it is a new version at all but it can be compared with the original changes of JT.

File: engine/uk.co.spudsoft.birt.emitters.excel/src/uk/co/spudsoft/birt/emitters/excel/StyleManagerUtils.java [hunk 1]
```diff
@@ -786,11 +787,12 @@ protected int getRichTextRunIndexForStart(List<RichTextRun> richTextRuns, int st
 	 * @param defaultFont  The font to be used prior to the first RichTextRun.
 	 * @param widthMM      The width of the output.
 	 * @param richTextRuns The list of RichTextRuns to be applied to the string
-	 * @return The heigh, in points, of a box big enough to contain the formatted
+	 * @param wrap         The text use wrapped style
```
Comment (hvbtup) [inline engine/uk.co.spudsoft.birt.emitters.excel/src/uk/co/spudsoft/birt/emitters/excel/StyleManagerUtils.java:790]: The comment for wrap is not clear to me (and grammar errors)

**Review by hvbtup — COMMENTED** [review]
There's too much code changed to understand it.
Is all of this actually related to formulas?

I mean, you added a whole line-breaking algorithm. 
From what I did decades ago when I developed the wordaxe hyphenation library including using it in a new a paragraph line-breaking algorithm for the Python ReportLab library, I know that those line-breaking algorithms are very fragile kind of program, where things can easily go wrong.

Is it possible for you to split the formula support and the layout enhancements you mentioned (which exactly?) into two separate PRs?

File: engine/uk.co.spudsoft.birt.emitters.excel/src/uk/co/spudsoft/birt/emitters/excel/StyleManagerUtils.java (hunk 1 above)
Comment (speckyspooky) [reply engine/uk.co.spudsoft.birt.emitters.excel/src/uk/co/spudsoft/birt/emitters/excel/StyleManagerUtils.java:790]: I will change it.

**Comment by speckyspooky** [comment]
Of course there are some changes because the biggest change is to add the "formula"-options and give the formula-values the option for the correct styling on excel side.

You talking about the "new a paragraph line-breaking algorithm" - which section do you mean exactly?
Then I will check if it make sense to split it from my side or may be we can use an additional user-property for the "new line algroithm" (default: false).

**Comment by hvbtup** [comment]
I mean the height calculation in `calculateTextHeightPoints`.

**Comment by hvbtup** [comment]
Thomas, if you say this is needed then I believe you :-).
It's just that I don't understand what's  the height calculation got to do with formulas...
My naive feeling is that when a cell contains a formula, then it's not possible to automatically compute a meaningful height anyway, because we don't know the value (only the formula).

I'm +0.5 for merging. Even if I don't understand all the changes, probably they will have an effect only for cells with formulas, which are a new feature. Thus I think that the risk of breaking existing reports is near zero.

**Pull request merged**